wkcuber/api/MagDataset.py

```diff
@@ -267,6 +280,64 @@ def _extract_file_index(self, file_path: str) -> Tuple[int, int, int]:
         zyx_index = [int(el[1:]) for el in file_path.split("/")]
         return zyx_index[2], zyx_index[1], zyx_index[0]
 
+    def compress(self, target_path: Path = None, args: Namespace = None) -> None:
+        # The data gets compressed inplace, if target_path is None.
+        # Otherwise it is written to target_path/layer_name/mag.
+
+        uncompressed_full_path = (
+            Path(self.layer.dataset.path) / self.layer.name / str(Mag(self.name))
+        )
+        compressed_path = (
+            target_path
+            if target_path is not None
+            else Path("{}.compress-{}".format(self.layer.dataset.path, uuid4()))
+        )
+        compressed_full_path = compressed_path / self.layer.name / str(Mag(self.name))
+
+        if compressed_full_path.exists():
+            logging.error(
+                "Target path '{}' already exists".format(compressed_full_path)
+            )
+            exit(1)
+
+        logging.info(
+            "Compressing mag {0} in '{1}'".format(
+                self.name, str(uncompressed_full_path)
+            )
+        )
+
+        was_opened = self.view._is_opened
+        if not was_opened:
+            self.open()  # opening the view is necessary to set the dataset
+        assert self.view.dataset is not None
+
+        # create empty wkw.Dataset
+        self.view.dataset.compress(str(compressed_full_path))
+
+        # compress all files to and move them to 'compressed_path'
+        with get_executor_for_args(args) as executor:
+            job_args = []
+            for file in self.view.dataset.list_files():
+                rel_file = Path(file).relative_to(self.layer.dataset.path)
+                job_args.append((Path(file), compressed_path / rel_file))
+
+            wait_and_ensure_success(
+                executor.map_to_futures(compress_file_job, job_args)
+            )
+
+        logging.info("Mag {0} successfully compressed".format(self.name))
+
+        if not was_opened:
+            self.close()
+
+        if target_path is None:
+            shutil.rmtree(uncompressed_full_path)
+            shutil.move(str(compressed_full_path), uncompressed_full_path)
+            shutil.rmtree(compressed_path)
```

Comment:
Can't you do this with a single move?

Reply:
`shutil.move` can only replace the target, if the target is a file. `uncompressed_full_path` is a directory, therefore, `shutil.rmtree(uncompressed_full_path)` is necessary. Deleting `compressed_path` is also necessary because otherwise this would leave an empty folder behind

Reply:
alright! thanks for clarifying